### ISSUE TITLE
Ability to get a previous node from the context

### ISSUE THREAD

**Issue by denis-sokolov** [eslint/eslint#3460]
I am writing a few rules catching common typos in multiple assignment statements. For example:

```js
foo.x = getX();
foo.x = getY(); 
```

Clearly this is a mistake. While I declare `ExpressionStatement` to be the scope of my rule, I need to "peek" to see the previous node and check whether it assigns to the same variable.

Currently `context` already provides `getTokenBefore`, `getNodeByRangeIndex`, and other convenience methods. It seems reasonable for it to have a `getNodeBefore` method.
I can almost simulate it with `getNodeByRangeIndex(node.range[0] - 1)`, but that's hacky and breaks if there are multiple whitespace characters between the nodes.

Is there a chance to see a `getNodeBefore` or a similar method on the context or in the ast-utils?

**Comment by eslintbot** [comment]
Thanks for the issue! We get a lot of issues, so this message is automatically posted to each one to help you check that you've included all of the information we need to help you.

Reporting a bug? Please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
1. The source code that caused the problem
1. The configuration you're using (for the rule or your entire config file)
1. The actual ESLint output complete with line numbers

Requesting a new rule? Please be sure to include:

1. The use case for the rule - what is it trying to prevent or flag?
1. Whether the rule is trying to prevent an error or is purely stylistic
1. Why you believe this rule is generic enough to be included

Requesting a feature? Please be sure to include:

1. The problem you want to solve (don't mention the solution)
1. Your take on the correct solution to problem

Including this information in your issue helps us to triage it and get you a response as quickly as possible.

Thanks!

**Comment by lo1tuma** [comment]
You could do the following:
```js
getNodeByRangeIndex(context.getTokenBefore(node).range[1])
```

**Comment by denis-sokolov** [comment]
That works if I use the 0-th range index, not the 1-st:

```js
getNodeByRangeIndex(context.getTokenBefore(node).range[0])
```

Thank you, @lo1tuma.


Still, this seems something that is beneficial to abstract away in a `getPreviousNode`, as opposed to copying across rules, isn't it?

**Comment by gyandeeps** [comment]
Do we need this enhancement or not?
ping @eslint/eslint-team

**Comment by nzakas** [comment]
I don't think we need it right now. If I'm wrong, we can reopen.
